crates/libtest2/src/macros.rs

```diff
@@ -38,6 +51,13 @@ macro_rules! _test_parse {
             fn run(&self, context: &$crate::TestContext) -> $crate::RunResult {
                 fn run $($item)*
 
+                $(
+                    match $crate::_private::parse_ignore!($($attr)*) {
+                        ::std::option::Option::None => context.ignore()?,
+                        ::std::option::Option::Some(reason) => context.ignore_for(reason)?,
+                    }
+                )*
+
```

Comment:
As an FYI, something I'm curious about is the idea of `ignore` being a special case of a general "test prelude" feature that allows test authors to run logic before the test.

Reply:
This "prelude" idea would be a bit different than setup/teardown since it couldn't pass parameters to the test.